src/patternfly/components/TextInputGroup/text-input-group-text-input.hbs

```diff
@@ -1,11 +1,25 @@
-<input class="pf-c-text-input-group__text-input{{~#if text-input-group-text-input--modifier}} {{text-input-group-text-input--modifier}}{{/if}}"
+<input class="pf-c-text-input-group__text-input
+  {{~#if text-input-group-text-input--hint-text}} pf-m-hint{{/if}}
+  {{~#if text-input-group-text-input--modifier}}
+    {{text-input-group-text-input--modifier}}
+  {{/if}}"
   type="text"
-  value="{{#if text-input-group--IsDisabled}}Disabled{{else}}{{text-input-group--value}}{{/if}}"
+  {{#if text-input-group--IsDisabled}}
+    value="Disabled"
+  {{else if text-input-group-text-input--hint-text}}
+    value="{{text-input-group-text-input--hint-text}}"
+  {{else}}
+    value="{{text-input-group--value}}"
+  {{/if}}
   aria-label="Type to filter"
```

Comment:
Can we omit this if it's a hint input? It won't do anything if it's `aria-hidden`. Looks like the react component is already doing that in the search input. We'll need to update the search input, too - we can do that here or in a separate issue if you'd prefer.

```suggestion
  {{#if text-input-group-text-input--hint-text }}
    aria-label="Type to filter"
  {{/unless}}
```